controllers/common_test.go

```diff
@@ -148,4 +158,42 @@ var _ = Describe("GetGeneratedJobName", func() {
 			Expect(uidWithoutHyphens).To(ContainSubstring(generatedJobName))
 		})
 	})
+
+	When("An additional postfix is required and maxNameLen is sufficiently large", func() {
+		BeforeEach(func() {
+			maxNameLen = 253
+			objectMetaName = "object.meta.name"
+			generatedPostfix = "generated.postfix"
+			shouldJobNameHavePostfix = true
+
+		})
+
+		It("Concatenates the name, shortened uid, generatedPostfix", func() {
+			Expect(generatedJobName).To(ContainSubstring(objectMetaName))
+			Expect(generatedJobName).To(ContainSubstring(uidWithoutHyphens))
+			Expect(generatedJobName).To(ContainSubstring(generatedPostfix))
+		})
+
+		It("Length does not exceed maxNameLen", func() {
+			Expect(len(generatedJobName)).To(BeNumerically("<=", maxNameLen))
+		})
+	})
+
+	When("An additional postfix is required and maxNameLen is not large enough for objectMetaName", func() {
+		BeforeEach(func() {
+			maxNameLen = 64
+			objectMetaName = strings.Repeat("A", 70)
+			generatedPostfix = "generated.postfix"
+			shouldJobNameHavePostfix = true
+		})
+
+		It("Concatenates the shortened uid, generatedPostfix", func() {
+			Expect(generatedJobName).To(ContainSubstring(uidWithoutHyphens))
+			Expect(generatedJobName).To(ContainSubstring(generatedPostfix))
+		})
+
+		It("Length does not exceed maxNameLen", func() {
+			Expect(len(generatedJobName)).To(BeNumerically("<=", maxNameLen))
+		})
+	})
```

Comment:
Is there a test for substrings of the required part without a required postfix? Just making sure we hit each condition

Reply:
without a required postfix - not sure what you mean. 
There is a test to ensure that if required is larger than the maxlen, uid is prioritizied.